### PR TITLE
add IsAreaOk function instead of multiple manual checks

### PR DESCRIPTION
**Pull request by qndel**

### CODE REVIEW

**Review by obligaron — APPROVED** [review]
lgtm 🙂

**Review by glebm — APPROVED** [review]

File: Source/objects.cpp [hunk 1]
```diff
@@ -321,6 +321,15 @@ bool RndLocOk(int xp, int yp)
 	return IsNoneOf(leveltype, DTYPE_CATHEDRAL, DTYPE_CRYPT) || dPiece[xp][yp] <= 125 || dPiece[xp][yp] >= 143;
 }
 
+bool IsAreaOk(Rectangle rect)
+{
+	for (Point position : PointsInRectangle(rect)) {
+		if (!RndLocOk(position.x, position.y))
+			return false;
+	}
+	return true;
```
Comment (glebm) [inline Source/objects.cpp:330]:
```suggestion
    const points = PointsInRectangle(rect);
    return std::any_of(points.begin(), points.end(), [](Point point) { return RndLocOk(point.x, point.y); });
```

(and `#include <algorithm>`)

Comment (glebm) [reply Source/objects.cpp:330]: I've add `c_all_of` in #6450, which will make the following possible:

``c++
return c_all_of(PointsInRectangle(rect), &RndLocOk);
```

Comment (AJenbo) [reply Source/objects.cpp:330]: Lets do that as a follow up?

**Pull request merged**